lib/quickbooks/model/active_record_scaffold.rb

```diff
@@ -0,0 +1,113 @@
+module Quickbooks
+  module Model
+    module ActiveRecordScaffold
+
+      def self.included(base)
+        base.extend(ClassMethods)
+      end
+
+      def save(options = {})
+        if id.blank?
+          self.class.send(:initialize_service).create(self, options)
+        else
+          self.class.send(:initialize_service).update(self, options)
+        end
+      end
+
+      module ClassMethods
+        def all(query = nil, options = {})
+          initialize_service(options).query(query, options).entries
+        end
+
+        def first
+          all.first
+        end
+
+        def where(options = {})
+          if options.is_a?(String)
+            all(build_string_query(options), {})
+          elsif options.present?
+            all(build_hash_query(options.except(:skip_pagination)), options)
+          else
+            all(full_query, {})
+          end
+        end
+
+        def find_by(options = {})
+          where(options).first
```

Comment:
seems like this should do a limit 1 to avoid fetching and parsing records that aren't used.